test/addons-napi/test_constructor/test_constructor.c

```diff
@@ -82,11 +82,11 @@ void Init(napi_env env, napi_value exports, napi_value module, void* priv) {
   if (status != napi_ok) return;
 
   napi_property_descriptor properties[] = {
-      { "echo", Echo, 0, 0, 0, napi_default, 0 },
-      { "accessorValue", 0, GetValue, SetValue, 0, napi_default, 0},
-      { "readwriteValue", 0, 0, 0, number, napi_default, 0 },
-      { "readonlyValue", 0, 0, 0, number, napi_read_only, 0},
-      { "hiddenValue", 0, 0, 0, number, napi_read_only | napi_dont_enum, 0},
+      { "echo", Echo, 0, 0, 0, napi_enumerable, 0 },
+      { "accessorValue", 0, GetValue, SetValue, 0, napi_enumerable, 0},
```

Comment:
The `accessorValue` is not actually referenced in the JS layer. You should also find a way to make certain the getters/setters are actually called when getting/setting the property.

Also, tests that cover the [normalization logic](https://github.com/jasongin/nodejs/blob/5d57259cb3d9be6559f8ba82748470c835fa0cac/src/node_api.cc#L782-L786) fully should be added.